File: app/src/main/java/com/chiranths/jobportal1/Adapters/CenterHomeadaptor.java
package com.chiranths.jobportal1.Adapters;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.chiranths.jobportal1.Activities.BasicActivitys.HotelsDetails;
import com.chiranths.jobportal1.Model.HotelsModel;
import com.chiranths.jobportal1.Model.ProductInfo;
import com.chiranths.jobportal1.Activities.HotDealsactivity.HotDealsDetailsActivity;
import com.chiranths.jobportal1.R;

import java.util.List;

public class CenterHomeadaptor extends RecyclerView.Adapter<CenterHomeadaptor.ViewHolder> {


    private List<HotelsModel> productInfos;
    private Context context;


    public CenterHomeadaptor(List<HotelsModel> productInfos, Context context) {
        this.productInfos = productInfos;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        context = parent.getContext();
        View listItem= layoutInflater.inflate(R.layout.recyclar_center_layout, parent, false);
        ViewHolder viewHolder = new ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {

        HotelsModel productInfo = productInfos.get(position);

        Glide.with(context)
                .load(productInfo.getImage())
                .into(holder.iv_image);

        holder.tv_name_hot.setText(productInfo.getName());
        holder.tv_final_price_hot.setText("Rent: "+productInfo.getPrice());
        holder.tv_hotel_website.setText(productInfo.getWebsite());
        holder.tv_advance.setText("Advance: ");
        holder.tv_loaction_hot.setText(productInfo.getAddress());

        holder.cv_deals.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent =new Intent(context, HotelsDetails.class);
                intent.putExtra("pid",productInfo.getPid());
                context.startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return productInfos.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView iv_image,iv_call_bottom,iv_whatsapp_bottom;
        CardView cv_deals;
        TextView tv_name_hot,tv_final_price_hot,tv_hotel_website,tv_loaction_hot,tv_advance;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            iv_image = itemView.findViewById(R.id.iv_location_image_center);
            cv_deals = itemView.findViewById(R.id.cv_card_deals_center);
            tv_name_hot = itemView.findViewById(R.id.tv_name_hot_center);
            tv_final_price_hot = itemView.findViewById(R.id.tv_final_price_hot_center);
            tv_hotel_website = itemView.findViewById(R.id.tv_hotel_website);
            tv_loaction_hot = itemView.findViewById(R.id.tv_loaction_hot_center);
            tv_advance = itemView.findViewById(R.id.tv_home_advance);
        }
    }
}

File: app/src/main/java/com/sbd/gbd/Adapters/CenterHomeadaptor.kt
package com.sbd.gbd.Adapters

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.database.FirebaseDatabase
import com.sbd.gbd.Model.HotelsModel
import com.sbd.gbd.R
import com.sbd.gbd.Utilitys.AppConstants
import com.sbd.gbd.Utilitys.Utilitys

class CenterHomeadaptor(private val productInfos: ArrayList<HotelsModel>, private var context: Context) :
    RecyclerView.Adapter<CenterHomeadaptor.ViewHolder>() {
    var utilitys = Utilitys()

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        context = parent.context
        val listItem =
            layoutInflater.inflate(R.layout.recyclar_center_layout, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, @SuppressLint("RecyclerView") position: Int) {
        val productInfo = productInfos[position]
        Glide.with(context)
            .load(productInfo.image)
            .into(holder.iv_image)
        holder.tv_name_hot.text = productInfo.name
        holder.tv_final_price_hot.text = productInfo.price
       // holder.tv_hotel_website.text = productInfo.website
        holder.tv_advance.text = productInfo.category
        holder.tv_loaction_hot.text = productInfo.address

        if (productInfo.status.equals("1")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else if (productInfo.status.equals("3")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else {
            holder.ll_approve.visibility = View.GONE
            holder.ll_call?.visibility = View.VISIBLE
        }

        holder.ll_approve.setOnClickListener {
            productInfo.pid?.let { it1 ->
                FirebaseDatabase.getInstance().reference.child("hotelsforyou").child(it1)
                    .child(AppConstants.Status).setValue("2")
            }
        }
        holder.iv_const_call_bottom.setOnClickListener { view: View? ->
            utilitys.navigateCall(
                context,
                productInfo.Number,
                productInfo.owner
            )
        }
        holder.iv_const_whatsapp_bottom.setOnClickListener { view: View? ->
            utilitys.navigateWhatsapp(
                context,
                productInfo.Number,
                productInfo.owner
            )
        }
    }

    override fun getItemCount(): Int {
        return productInfos.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var iv_image: ImageView
        var iv_call_bottom: ImageView? = null
        var iv_whatsapp_bottom: ImageView? = null
        var cv_deals: CardView
        var tv_name_hot: TextView
        var tv_final_price_hot: TextView
        var tv_hotel_website: TextView
        var tv_loaction_hot: TextView
        var tv_advance: TextView
        var ll_approve: LinearLayout
        var ll_call: LinearLayout
        var iv_const_whatsapp_bottom: ImageView
        var iv_const_call_bottom: ImageView

        init {
            iv_image = itemView.findViewById(R.id.iv_location_image_center)
            cv_deals = itemView.findViewById(R.id.cv_card_deals_center)
            tv_name_hot = itemView.findViewById(R.id.tv_name_hot_center)
            tv_final_price_hot = itemView.findViewById(R.id.tv_final_price_hot_center)
            tv_hotel_website = itemView.findViewById(R.id.tv_hotel_website)
            tv_loaction_hot = itemView.findViewById(R.id.tv_loaction_hot_center)
            tv_advance = itemView.findViewById(R.id.tv_home_advance)
            ll_approve = itemView.findViewById(R.id.ll_approve)
            ll_call = itemView.findViewById(R.id.ll_call)
            iv_const_whatsapp_bottom = itemView.findViewById(R.id.iv_whatsapp_bottom)
            iv_const_call_bottom = itemView.findViewById(R.id.iv_call_bottom)
        }
    }
}
